component/identity/src/main/java/org/gatein/portal/idm/impl/store/attribute/ExtendedAttributeManager.java

```diff
@@ -0,0 +1,131 @@
+/*
+ * This file is part of the Meeds project (https://meeds.io/).
+ * Copyright (C) 2023 Meeds Association
+ * contact@meeds.io
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.gatein.portal.idm.impl.store.attribute;
+
+import org.exoplatform.container.ExoContainer;
+import org.exoplatform.container.ExoContainerContext;
+import org.exoplatform.services.log.ExoLogger;
+import org.exoplatform.services.log.Log;
+import org.exoplatform.services.organization.idm.PicketLinkIDMService;
+import org.picketlink.idm.api.Attribute;
+import org.picketlink.idm.api.CredentialEncoder;
+import org.picketlink.idm.api.User;
+import org.picketlink.idm.common.exception.IdentityException;
+import org.picketlink.idm.impl.api.PasswordCredential;
+import org.picketlink.idm.impl.api.session.IdentitySessionImpl;
+import org.picketlink.idm.impl.api.session.managers.AttributesManagerImpl;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class ExtendedAttributeManager extends AttributesManagerImpl {
+
+  private static final Log            LOG                              = ExoLogger.getExoLogger(ExtendedAttributeManager.class);
+
+  public static final String          OLD_PASSWORD_SALT_USER_ATTRIBUTE = "passwordSalt";
+
+  public static final String          PASSWORD_SALT_USER_ATTRIBUTE     = "passwordSalt128";
+
+  private static final String         DEFAULT_ENCODER                  =
+                                                      "org.exoplatform.web.security.hash.Argon2IdPasswordEncoder";
+
+  private static final String         OLD_ENCODER_CLASS_PROPERTY       = "oldCredentialEncoder.class";
+
+  private static PicketLinkIDMService idmService;
+
+  public ExtendedAttributeManager(IdentitySessionImpl session) {
+    super(session);
+  }
+
+  @Override
+  public boolean validatePassword(User user, String password) throws IdentityException {
+    Attribute salt128 = getAttribute(user.getKey(), PASSWORD_SALT_USER_ATTRIBUTE);
```

Comment:
We don't need to add the length of salt in variable names for better future code upgradability

```suggestion
    Attribute salt = getAttribute(user.getKey(), PASSWORD_SALT_USER_ATTRIBUTE);
```

Reply:
Done